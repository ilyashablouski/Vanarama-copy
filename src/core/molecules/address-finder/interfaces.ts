import { IBaseProps } from '../../interfaces/base';

export interface IAddressSuggestion {
  id: string;
  label: string;
}

export interface IAddressFinderProps extends IBaseProps {
  /**
   * The API key for Loqate
   */
  apiKey: string;
  /**
   * Called when the user selects an address suggestion
   */
  onSuggestionChange: (suggestion?: IAddressSuggestion) => void;
  /**
   * Sets the currently selected suggestion
   */
  selected?: IAddressSuggestion;
}
