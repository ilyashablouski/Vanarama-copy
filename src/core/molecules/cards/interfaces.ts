import { CSSProperties } from 'react';
import { IBaseProps } from '../../interfaces/base';
import { ICardHeaderProps } from './CardHeader';
import { ICardTitleProps } from './CardTitle';

import { ICloudflareOptimisation } from '../../../helpers/imageOptimiseUtils/interfaces';
import { Component } from '../../../types/common';

export interface ICardProps extends IBaseProps {
  /**
   * Add "-inline" className
   */
  inline?: boolean;
  /**
   * Add "-overflow" className
   */
  overflow?: boolean;
  /**
   * Card header
   */
  header?: ICardHeaderProps;
  /**
   * Card image
   */
  imageSrc?: string;
  /**
   * Card title
   */
  title?: ICardTitleProps;
  /**
   * Card description
   */
  description?: string;
  /**
   * Image alt
   */
  alt?: string;
  /**
   * Inline style
   */
  style?: CSSProperties;
  /**
   * Image Placeholder
   */
  placeholderImage?: string;

  initialRentalDataTestId?: string;

  controlLengthDataTestId?: string;

  annualMileageDataTestId?: string;

  annualMileageBoosterDataTestId?: string;

  damageCoverDataTestId?: string;

  maintenanceDataTestId?: string;

  fuelDataTestId?: string;

  transmissionDataTestId?: string;

  colorDataTestId?: string;

  trimDataTestId?: string;

  descriptionDataTestId?: string;

  optimisedHost?: string;

  optimisationOptions?: ICloudflareOptimisation;

  loadImage?: boolean;

  lazyLoad?: boolean;

  loadImageProps?: (options?: any) => void;

  extrasRender?: Component | null;

  isBlackFridayFlag?: boolean;
}
