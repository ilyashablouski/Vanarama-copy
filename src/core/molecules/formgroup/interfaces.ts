import { IBaseProps } from '../../interfaces/base';

export interface IFormgroupProps extends IBaseProps {
  children?: React.ReactNode;
  /**
   * The ID of the control this is wrapping
   */
  controlId?: string;
  /**
   * An optional error message to display.
   */
  error?: string | React.ReactNode;
  /**
   * Use hint text for help thatâ€™s relevant to the majority of users, like how their information will be used, or where to find it.
   */
  hint?: string;
  /**
   * Whether to display the children inline
   */
  inline?: boolean;
  /**
   * A label for the inputs
   */
  label?: string;
  /**
   * Whether to display the children editable
   */
  editable?: string;
}
