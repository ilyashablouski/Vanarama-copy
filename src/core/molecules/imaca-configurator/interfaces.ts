export interface IColor {
  names?: string;
  hex: string;
}

export interface IImacaConfigurator {
  id: string;
  className?: string;
  width: number;
  height: number;
  onClick?: () => void;
  onMouseDown?: () => void;
  assets: {
    colors: IColor[];
    vehicleUrl: string;
    rimsUrl: string;
    tyresUrl: string;
  };
}
