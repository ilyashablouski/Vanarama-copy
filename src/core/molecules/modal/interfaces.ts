import { IBaseProps } from '../../interfaces/base';

export interface IModalProps extends IBaseProps {
  title?: string;
  text?: string;
  show: boolean;
  additionalText?: string;
  onRequestClose?: () => void;
  containerClassName?: string;
  dataUiTestIdHeading?: string;
}
