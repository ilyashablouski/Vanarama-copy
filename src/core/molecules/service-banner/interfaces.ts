export interface IServiceBannerLink {
  url?: string | null;
  text?: string | null;
}

export interface IServiceBanner {
  enable?: boolean | null;
  message?: string | null;
  link?: IServiceBannerLink | null;
  className?: string;
}
