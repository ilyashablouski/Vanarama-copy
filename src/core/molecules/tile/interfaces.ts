import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';

export interface ITileProps extends IBaseProps {
  color?: TColor;
  centered?: boolean;
  scrollable?: boolean;
  plain?: boolean;
}
