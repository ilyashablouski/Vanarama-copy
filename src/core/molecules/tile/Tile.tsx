import React, { FC, memo } from 'react';
import cx from 'classnames';

import { ITileProps } from './interfaces';

const Tile: FC<ITileProps> = memo(props => {
  const {
    className,
    children,
    centered,
    color,
    dataTestId,
    scrollable,
    plain,
  } = props;

  return (
    <div
      className={cx('tile', className, {
        [`-${color} tile--color`]: color,
        '-scrollable': scrollable,
        '-plain': plain,
        'tile--centered': centered,
      })}
      data-testid={dataTestId}
    >
      {children}
    </div>
  );
});

Tile.displayName = 'Tile';

export default Tile;
