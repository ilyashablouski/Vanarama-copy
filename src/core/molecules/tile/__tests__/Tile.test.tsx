import React from 'react';
import renderer from 'react-test-renderer';

import Tile from '..';
import { ITileProps } from '../interfaces';

const optionalProps = {
  color: 'orange' as const,
  scrollable: true,
};

function getComponent(props?: ITileProps) {
  return renderer
    .create(
      <Tile {...props}>
        <p>Child 1</p>
        <p>Child 2</p>
        <p>Child 3</p>
      </Tile>,
    )
    .toJSON();
}

describe('<Tile />', () => {
  it('renders correctly', () => {
    const tree = getComponent();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with optional props', () => {
    const tree = getComponent(optionalProps);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when centered', () => {
    const tree = getComponent({ centered: true });
    expect(tree).toMatchSnapshot();
  });
});
