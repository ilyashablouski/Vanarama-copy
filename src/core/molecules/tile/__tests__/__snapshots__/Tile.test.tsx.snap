// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile /> renders correctly 1`] = `
<div
  className="tile"
>
  <p>
    Child 1
  </p>
  <p>
    Child 2
  </p>
  <p>
    Child 3
  </p>
</div>
`;

exports[`<Tile /> renders correctly when centered 1`] = `
<div
  className="tile tile--centered"
>
  <p>
    Child 1
  </p>
  <p>
    Child 2
  </p>
  <p>
    Child 3
  </p>
</div>
`;

exports[`<Tile /> renders correctly with optional props 1`] = `
<div
  className="tile -orange tile--color -scrollable"
>
  <p>
    Child 1
  </p>
  <p>
    Child 2
  </p>
  <p>
    Child 3
  </p>
</div>
`;
